inc/user.class.php

```diff
@@ -497,11 +497,16 @@ function isEmail($email) {
     * Retrieve an item from the database using its personal token
     *
     * @param $token user token
+    * @param string $field the field storing the token
     *
     * @return true if succeed else false
    **/
-   function getFromDBbyToken($token) {
-      return $this->getFromDBByQuery("WHERE `".$this->getTable()."`.`personal_token` = '$token'");
+   function getFromDBbyToken($token, $field = 'personal_token') {
+      if (!in_array($field, ['personal_token', 'api_token'])) {
+         return false;
```

Comment:
Maybe add something in the logs in this case? This sounds like an error to request something that does not exists.